### ISSUE TITLE
[REVIEW]: GIBBON: The Geometry and Image-Based Bioengineering add-ON

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#506]
**Submitting author:** @Kevin-Mattheus-Moerman (<a href="http://orcid.org/0000-0003-3768-4269">Kevin Moerman</a>)
**Repository:** <a href="https://github.com/gibbonCode/GIBBON" target ="_blank">https://github.com/gibbonCode/GIBBON</a>
**Version:** v3.0.0
**Editor:** @cMadan
**Reviewer:** @siboles
**Archive:** <a href="http://dx.doi.org/10.5281/zenodo.1175348" target="_blank">10.5281/zenodo.1175348</a>

## Status

[![status](https://joss.theoj.org/papers/7666e082975be31b8a64b5fe1e64f940/status.svg)](https://joss.theoj.org/papers/7666e082975be31b8a64b5fe1e64f940)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/7666e082975be31b8a64b5fe1e64f940"><img src="https://joss.theoj.org/papers/7666e082975be31b8a64b5fe1e64f940/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/7666e082975be31b8a64b5fe1e64f940/status.svg)](https://joss.theoj.org/papers/7666e082975be31b8a64b5fe1e64f940)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/gibbonCode/GIBBON" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@siboles, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @cMadan know.

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/gibbonCode/GIBBON">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (v2.5.0)?
- [x] **Authorship:** Has the submitting author (@Kevin-Mattheus-Moerman) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks. @siboles it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
```
https://github.com/openjournals/joss-papers/blob/joss.00506/joss.00506/10.21105.joss.00506.pdf
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@cMadan will you add an additional checklist for @wspr? I did this [here](https://github.com/openjournals/joss-reviews/issues/500#issuecomment-351155247) by simply manually editing the tick box comment and coping in an additional tick box list. You also need to invite the second reviewer (ping @arfon for that) so they can be assigned to the issue and tick boxes.

**Comment by cMadan** [comment]
Thanks for also agreeing to review, @wspr! Below is the reviewer checklist for you to fill out.

### Conflict of interest

- [] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/gibbonCode/GIBBON">repository url</a>?
- [] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [] **Version:** Does the release version given match the GitHub release (v2.5.0)?
- [] **Authorship:** Has the submitting author (@Kevin-Mattheus-Moerman) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by Kevin-Mattheus-Moerman** [comment]
@wspr let me know if I can help with anything. See `testGibbon` to run all tests.

**Comment by cMadan** [comment]
@siboles, is it safe to assume that you are now satisfied with the submission and endorse it's acceptance in the current form?

@Kevin-Mattheus-Moerman, I'll still wait for @wspr, but thought I'd get that confirmation from @siboles, rather than checking later on.

**Comment by wspr** [comment]
Hi all — the end of the year hit me rather suddenly so I haven’t had a chance to fully investigate. If you could wait a little longer (say, before the end of the year) I’d be very happy to put in my 2c as well, but if you’d rather knock it on the head immediately I am happy to give it a miss. Sorry for the hold up.

**Comment by cMadan** [comment]
@wspr, I have no problem waiting a bit longer. Also feel free to take some time into January too (I'd hate to have reviewing take away from holiday plans).

That said, can you check if that second checklist works for you? If not, I'll have to see about permissions...

**Comment by wspr** [comment]
@cMadan Checklist doesn’t work I’m afraid — if it’s a hassle I can just paste in my own :-)

**Comment by arfon** [comment]
@wspr - if you accept the invite link here: https://github.com/openjournals/joss-reviews/invitations you should be able to edit the checklist.

**Comment by siboles** [comment]
Very sorry folks for the slow reply. Yes I am satisfied with GIBBON and
vote to accept it for publication.

**Comment by cMadan** [comment]
No worries, @siboles! Thanks again for the review and enjoy the holidays :)

**Comment by arfon** [comment]
@Kevin-Mattheus-Moerman - At this point could you make an archive of the reviewed software in Zenodo/figshare/other service and update this thread with the DOI of the archive? I can then move forward with accepting the submission.

**Comment by arfon** [comment]
@Kevin-Mattheus-Moerman - ignore that. We're waiting for a second review here right?

**Comment by Kevin-Mattheus-Moerman** [comment]
@arfon Yes indeed. @wspr let me know if you need any help, looking forward to your comments.

**Comment by cMadan** [comment]
@wspr, do you still have time/interest in doing a review of this submission? Please let me know either way.

**Comment by Kevin-Mattheus-Moerman** [comment]
I've been working on @wspr 's comments. He had opened some issues on my repo which I believe I've addressed. There may be more to come but I'll let him comment on that.

**Comment by wspr** [comment]
@cMadan — sorry this dragged on a little; I think I'm basically there.

@Kevin-Mattheus-Moerman — did you see my last message for testing out the `git update-index --chmod=+x lib_ext/tetGen/mac64/tetgen` thing? I believe if you run that on your end then commit, I'll be able to check and confirm. The one thing I'm not sure about is how many more of these are needed; I guess it can't hurt to also run on the `lin64` script as well.

**Comment by cMadan** [comment]
Thank you for the review, @wspr! I think you're all done.

@Kevin-Mattheus-Moerman, let me know when you've made these minor changes to the paper. After that's done, please let me know the DOI for the archive.

**Comment by cMadan** [comment]
@Kevin-Mattheus-Moerman, just thought I'd check in--are you still working on @wspr's suggestions? No rush, just wanted to make sure you weren't waiting on something from me :)

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00506/joss.00506/10.21105.joss.00506.pdf)

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00506/joss.00506/10.21105.joss.00506.pdf)

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00506/joss.00506/10.21105.joss.00506.pdf)

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00506/joss.00506/10.21105.joss.00506.pdf)

**Comment by Kevin-Mattheus-Moerman** [comment]
Sorry for spamming this thread. I had some minor paper edits and things to iron out.

**Comment by Kevin-Mattheus-Moerman** [comment]
@cMadan I believe I have addressed all of @siboles and @wspr comments. 

@siboles and @wspr thanks for the thorough and useful review process! 🚀 🤖 🍰 

FYI All functionality which was reviewed remains. I do want to report some new features which I'll highlight briefly. 

* I upgraded GIBBON to also work with febio_spec 2.5 since the old functionality only covered febio_spec 2.0. New demos featuring febio_spec 2.5 are named `DEMO_febio_...`. The old demos using febio_spec 2.0 still work and remain named  `DEMO_FEBio...`.
* I implemented a new way to create the input file to allow for all of FEBio's functionality to be covered. (see `HELP_febioStruct2xml`). It is essentially a flexible way to handle XML files. So now any XML content can be coded with these tools allowing users to also code input files for experimental/novel FEBio features. This way GIBBON become ready not only for febio_spec 2.5 but also for any other future XML based release.

**Comment by Kevin-Mattheus-Moerman** [comment]
@cMadan Reviewer 2 (@wspr) commented about the fix for this issue that was raised https://github.com/gibbonCode/GIBBON/issues/17 it appears that is working now so I closed the issue. 

I also implemented the suggested changes to the paper proposed in this issue https://github.com/gibbonCode/GIBBON/issues/20. I went ahead and closed the issue but perhaps @wspr can comment if my changes are deemed sufficient.

**Comment by wspr** [comment]
I am very happy, indeed!

**Comment by cMadan** [comment]
@Kevin-Mattheus-Moerman, it looks like we're all set then! All that's left is the DOI for the archived code.

@siboles and @wspr, thank you again for reviewing this submission!

**Comment by Kevin-Mattheus-Moerman** [comment]
@cmadan great! 🎉 Here is the DOI of the archived version:    
[`https://doi.org/10.5281/zenodo.1175348`](https://doi.org/10.5281/zenodo.1175348)

@arfon the software version should be updated to be **v3.0.0**

@siboles @wspr thanks for the review! 🚀

**Comment by wspr** [comment]
Congrats! And thanks all, it was great to be involved… please keep me in mind if there are future reviewing opportunities.

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.1175348 as archive

**Comment by whedon** [comment]
OK. <a href="http://dx.doi.org/10.5281/zenodo.1175348" target="_blank">10.5281/zenodo.1175348</a> is the archive.

**Comment by arfon** [comment]
@wspr @siboles - many thanks for your reviews here and to @cMadan for editing this submission ✨ 

@Kevin-Mattheus-Moerman - your paper is now accepted into JOSS and your DOI is https://doi.org/10.21105/joss.00506 ⚡️ 🚀 💥

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippet:

```
[![DOI](http://joss.theoj.org/papers/10.21105/joss.00506/status.svg)](https://doi.org/10.21105/joss.00506)
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.00506/status.svg)](https://doi.org/10.21105/joss.00506)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://www.flipcause.com/secure/cause_pdetails/Mjk3Nzk=
